crates/oxc_linter/src/rules/unicorn/throw_new_error.rs

```diff
@@ -143,5 +145,10 @@ fn test() {
         ("throw (( getGlobalThis().Error ))()", None),
     ];
 
-    Tester::new(ThrowNewError::NAME, pass, fail).test_and_snapshot();
+    let fix = vec![
+        ("throw Error()", "throw new Error()"),
+        ("throw (( getGlobalThis().Error ))()", "throw new (( getGlobalThis().Error ))()"),
```

Comment:
Can you add some more cases that use member expressions and/or `Function.prototype.apply`?

Reply:
i don't think we report errors on this case currently? perhaps we should create a new ticket to track this rather than in this PR?